Adds info about client side range request (#23310)
cloudflare/cloudflare-docs: src/content/docs/cache/concepts/default-cache-behavior.mdx

When [Origin Cache Control](/cache/concepts/cache-control/) is enabled on an Enterprise customer’s website, it indicates that Cloudflare should strictly respect `Cache-Control` directives received from the origin server. Free, Pro and Business customers have this feature enabled by default. For a list of directives and behaviors when Origin Cache-Control is enabled or disabled, refer to [Cache-Control directives](/cache/concepts/cache-control/#cache-control-directives).

## Client side range requests

> Cloudflare supports clients to send range requests to be served from the cache. For a cacheable request, if the origin response includes a `Content-Length` header, then the specified byte range will be returned with a HTTP 206 response. If it does not include the `Content-Length` header, the cache will return the full content with a HTTP 200 response.

Reviewer: ```suggestion
Clients can send range requests to be served from the cache. For a cacheable request:

 - If the origin response includes a `Content-Length` header, then the specified byte range will be returned with a HTTP 206 response.
 - If the origin response does not include the `Content-Length` header, the cache will return the full content with a HTTP 200 response.
```
